cmd/cli/config/list.go

```diff
@@ -0,0 +1,74 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/jedib0t/go-pretty/v6/text"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/bacalhau-project/bacalhau/cmd/util/flags/cliflags"
+	"github.com/bacalhau-project/bacalhau/cmd/util/output"
+)
+
+func newListCmd() *cobra.Command {
+	o := output.OutputOptions{
+		Format:     output.TableFormat,
+		Pretty:     true,
+		HideHeader: false,
+		NoStyle:    false,
+		Wide:       false,
+	}
+	listCmd := &cobra.Command{
+		Use:   "list",
+		Short: "List all config keys.",
+		Args:  cobra.MinimumNArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return list(cmd, o)
+		},
+	}
```

Comment:
One more thing, please add the `cmd.util.ClientPreRunHooks` and `cmd.util.ClientPostRunHooks` to all of these client commands.

Reply:
huh, hadn't realized we needed these hooks on every command. Could we eventually move them to the root commands persistant pre run hook? Regardless will add them here.

Reply:
addressed.

Reply:
They are hooks for "client" commands only, i.e. not `serve` or `devstack`. For that reason they can't be on the root.